tests/integration/testlib/vxlan.py

```diff
@@ -0,0 +1,102 @@
+#
+# Copyright (c) 2019 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from contextlib import contextmanager
+
+import libnmstate
+
+from libnmstate.schema import Interface
+from libnmstate.schema import InterfaceState
+from libnmstate.schema import VXLAN
+
+
+def vxlans_up(vxlans):
+    vxlans_state = []
+    for vxlan in vxlans:
+        vxlans_state.append(vxlan_up(*vxlan))
+    return {Interface.KEY: vxlans_state}
+
+
+def vxlans_absent(vxlans):
+    vxlans_state = []
+    for vxlan in vxlans:
+        vxlans_state.append(vxlan_absent(*vxlan))
+    return {Interface.KEY: vxlans_state}
+
+
+@contextmanager
+def apply_state(setup_state, teardown_state):
+    try:
+        libnmstate.apply(setup_state)
+        yield setup_state
+    finally:
+        libnmstate.apply(teardown_state, verify_change=False)
+
+
+@contextmanager
+def vxlan_interfaces(vxlans):
+    vxlans_interfaces_absent = []
+    try:
+        setup_state = vxlans_up(vxlans)
+        libnmstate.apply(setup_state)
+        yield setup_state
+    finally:
+
+        # We need a little adaptation to remove remote from the tuple
+        # since it's not needed for absent
+        for vxlan in vxlans:
+            vxlan_id, parent, _ = vxlan
+            vxlans_interfaces_absent.append((vxlan_id, parent))
+
+        libnmstate.apply(
+            vxlans_absent(vxlans_interfaces_absent), verify_change=False
+        )
+
+
+def vxlan_up(vxlan_id, base_if, remote):
```

Comment:
How about `create_vxlan_state`?

Reply:
create_vxlans would be since name looks like an adjetive instead of verb.

Reply:
After the OOP refactor it's just up_state since it's a simple factory to genereate the state